nmostesting/mocks/Registry.py

```diff
@@ -168,6 +192,100 @@ def check_authorized(self, headers, path, write=False):
                 return 400
         return True
 
+    # Query API subscription support methods
+
+    def subscribe_to_query_api(self, version, resource_path):
+        """creates a subscription and starts a Subscription WebSocket"""
+
+        resource_type = self._get_resource_type(resource_path)
+
+        resource_types = ['node', 'device', 'source', 'flow', 'sender', 'receiver']
+
+        if resource_type not in resource_types:
+            raise SubscriptionException("Unknown resource type:" + resource_type
+                                        + " from resource path:" + resource_path)
+
+        # return existing subscription for this resource type if it already exists
+        if resource_type in self.subscriptions:
```

Comment:
Agreed - now returns 501 code for any subscription request not supported by the current mock implementation - still works with Riedel explorer

Reply:
I saw your commit c1ad995. I'm not sure it goes quite far enough... Did you consider subsequent `POST` requests for subscriptions of the same type, and whether the values of the current subscription match the new request? I know that currently you're not actually storing most of the values, because `params` and `secure` only support fixed values right now, and that's probably also true for `authorization` even though that isn't being checked yet, so maybe it's just `max_update_rate_ms`... but see also my other comment about implementing `GET /subscriptions` and `GET /subscriptions/{subscriptionId}`.

Reply:
Need to discuss further the how permissive/restrictive the subscription mechanism should be to satisfy the requirements of the tests